test/integration/sidekiq_test.rb

```diff
@@ -39,6 +39,19 @@ class SidekiqIntegrationTest < ActiveSupport::TestCase
     assert_equal 0, queue_size
   end
 
+  test "Time cursor corruption is prevented" do
```

Comment:
Could we add a test for Resque as well? Or does resque do not convert a Time cursor into a JSON?

Reply:
I had only put this test in as a kind of "regression" test for the bug we experienced, but I can look at adding it to Resque's tests too. 👍